### PR TITLE
E2e tests Feature 3

### PR DESCRIPTION
**Pull request by andriizhegurov-okta**
Actual script used or Future 3:

```
Feature: 3.1: Direct Auth Password Recovery

  Background:
	Given an org with an ORG Policy that defines Authenticators with Password and Email as required
	And a user named "Mary"
	And Mary is a user with a verified email and a set password

  Scenario: 3.1.1 Mary resets her password
	Given Mary navigates to the Self Service Password Reset View
	When she inputs her correct Email
	And she submits the recovery form
	Then she sees a page to select an authenticator
	When she chooses Email
	And she submits the select form
	Then she sees a page to input her code
	When she fills in the correct code
	And she submits the verification form
	Then she sees a page to set her password
	When she fills a password that fits within the password policy
	And she confirms that password
	And she submits the change password form
	Then she is redirected to the Root Page is provided
	
  Scenario: 3.1.2 Mary tries to reset a password with the wrong email
	Given Mary navigates to the Self Service Password Reset View
	Then she sees the Password Recovery Page
	When she inputs an Email that doesn't exist
	And she submits the recovery form
	Then she sees a message "There is no account with the Username {username}." 
```

Also:
[+] TerminalStateException 
[-] AuthenticateAsync.Terminal

### CODE REVIEW

File: src/Okta.Idx.Sdk.E2ETests/Helpers/OktaSdkHelper.cs [hunk 1]
```diff
@@ -34,6 +34,11 @@ public async Task<IUser> CreateActiveUserIdentifiedWithPasswordAsync(string emai
             return await _client.Users.CreateUserAsync(options);
         }
 
+        public Task<IUser> CreateUnassignedUserIdentifiedWithPasswordAsync(string userName, string password)
+        {
+            throw new NotImplementedException();
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Helpers/OktaSdkHelper.cs:39]: I guess we're not using this method for 3.1 right?

File: src/Okta.Idx.Sdk.E2ETests/Helpers/TestUserHelper.cs [hunk 2]
```diff
@@ -38,15 +40,59 @@ public async Task<TestUserProperties> GetActivePasswordUserAsync()
             };
         }
 
-        public TestUserProperties GetUnassignedUser()
+        public async Task<TestUserProperties> GetUnassignedUser()
         {
+            await CleanUpAsync();
+
+            var oktaUser = await _oktaHelper.CreateUnassignedUserIdentifiedWithPasswordAsync(_a18nProfile.EmailAddress, _configuration.UserPassword);
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Helpers/TestUserHelper.cs:47]: hmm `CreateUnassignedUserIdentifiedWithPasswordAsync` is not implemented, what am I missing?

File: src/Okta.Idx.Sdk.E2ETests/Helpers/TestUserHelper.cs [hunk 3]
```diff
@@ -38,15 +40,59 @@ public async Task<TestUserProperties> GetActivePasswordUserAsync()
...
-                Password = _configuration.UserPassword
+                Email = oktaUser.Profile.Email,
+                PhoneNumber = oktaUser.Profile.PrimaryPhone,
+                Password = _configuration.UserPassword,
             };
         }
 
+        public async Task<string> GetRecoveryCodeFromEmail()
+        {
+            for (int tries = 0; tries < 30; tries++) {
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Helpers/TestUserHelper.cs:59]: did we choose 30 for any particular reason to retry?

File: src/Okta.Idx.Sdk.E2ETests/Helpers/OktaSdkHelper.cs (hunk 1 above)
Comment (andriizhegurov-okta) [reply src/Okta.Idx.Sdk.E2ETests/Helpers/OktaSdkHelper.cs:39]: Yes, we don't need this for this feature

File: src/Okta.Idx.Sdk.E2ETests/Helpers/TestUserHelper.cs [hunk 4]
```diff
@@ -38,15 +40,59 @@ public async Task<TestUserProperties> GetActivePasswordUserAsync()
...
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+        }
+
+        private string ExtractRecoveryCodeFromEmail(string mailContent)
+        {
+            var res = string.Empty;
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Helpers/TestUserHelper.cs:84]: Can we please use a more meaningful name for vars?

File: src/Okta.Idx.Sdk.E2ETests/Helpers/TestUserHelper.cs (hunk 2 above)
Comment (andriizhegurov-okta) [reply src/Okta.Idx.Sdk.E2ETests/Helpers/TestUserHelper.cs:47]: It's not for Feature 3, I started to implement next feature scenarios here.

File: src/Okta.Idx.Sdk.E2ETests/Helpers/TestUserHelper.cs (hunk 3 above)
Comment (andriizhegurov-okta) [reply src/Okta.Idx.Sdk.E2ETests/Helpers/TestUserHelper.cs:59]: No, it's my guess that this will be enough. Usually mail comes in a couple of seconds.

**Review by laura-rodriguez — APPROVED** [review]
Minor comments, LGTM.

**Pull request merged**